locale/import-symbols.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3.5
+# coding: utf-8
+# NVDA symbols and cldr import
+# Import NVDA symbols dictionaries then emojies from CLDR database
+# Copyright (C) 2001, 2019 Brailcom, o.p.s.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+###
+#This file includes portions of code from the NVDA project.
+#URL: http://www.nvda-project.org/
+###
+
+from sys import argv as arguments, stderr
+from os import path
+import codecs
+from collections import OrderedDict
+import csv
+
+class SymbolsDictionary(OrderedDict):
+	""" This class inherited from OrderedDict is displayed as a .dic file content when converted to a string
+	Attribute: symbolsQualifier, default None - if the 4th column must have one """
+
+	def __init__(self, symbolsQualifier=None,*args,**kwargs):
+		super(SymbolsDictionary, self).__init__(*args,**kwargs)
+		self.symbolsQualifier=symbolsQualifier
+
+	def __str__(self):
+		dicText = str("symbols:\r\n")
+		for pattern, description in self.items():
+			if self.symbolsQualifier:
+				dicText = str().join((dicText,u"{pattern}\t{description}\tnone\t{qualifier}\r\n".format(
+				pattern=pattern,
+				description=description,
+				qualifier=self.symbolsQualifier
+				)))
+			else:
+				dicText = str().join((dicText,u"{pattern}\t{description}\tnone\r\n".format(
+				pattern=pattern,
+				description=description
+				)))
+		return dicText
+
+def createCLDRAnnotationsDict(sources):
+	""" Argument: the list of CLDR files to parse
+	Returns: a SymbolsDictionary object containing all emojies in the specified language """
+	from xml.etree import ElementTree
+	cldrDict = SymbolsDictionary() # Where the list of emojies will be storred
+	# We brows the source files
+	for source in sources:
+		tree = ElementTree.parse(source) # We parse the CLDR XML file to extract all emojies
+		for element in tree.iter("annotation"):
+			if element.attrib.get("type") == "tts":
+				cldrDict[element.attrib['cp']] = element.text.replace(":","")
+	assert cldrDict, "cldrDict is empty"
+	return cldrDict
+
+# The list of languages supported by Speech-Dispatcher and the corresponding language(s) in the CLDR database
```

Comment:
There is not really any "list of languages supported by Speech-Dispatcher", it's up to the speech syntheses. Whenever espeak-ng adds a new language, it'll immediately work in speech-dispatcher, without any change there. So we shouldn't have any list, and just pick put everything we can from CLDR, e.g. af, am, etc. (af for instance does happen to be supported by espeak-ng)

Reply:
And the same for NVDA symbols: there is no reason to avoid importing the NVDA symbols file even if CLDR doesn't provide emoji names. Just import everything that is availble from NVDA.